miio/miioprotocol.py

```diff
@@ -141,6 +142,9 @@ def send(self, command: str, parameters: Any = None, retry_count=3) -> Any:
 
         cmd = {"id": self._id, "method": command}
 
+        if self.sid is not None:
```

Comment:
I would much prefer to add a new kwarg-only option `extra_params` instead of hardcoding sid here, something like this: https://github.com/rytilahti/python-miio/commit/57ff02898dc34f516705ab65c56c6524f71e6c8b#diff-91c92bb750c47e7a1d365652c76443edR102 – `send()` inside gateway could then be overridden to pass the sid parameter where needed.

What do you think?

Reply:
Yes, you are right. It's just that I'm new to python, I'm just figuring out the syntax as I go. I didn't want to change the method's signature for this fringe feature, but if you have an IDE where you can do this refactoring easily, please do. I'm working with text editor here.

Reply:
Ok, I'll update this PR when I get some time this weekend. Thanks :-)

Reply:
Hi, I created a separate PR to add the necessary functionality (#653). Could you please rebase this on top of it? Thanks!

Reply:
Okay done. Thanks!

Reply:
Thanks! I think this will need to wait for a bit, I'll get that linked PR done and merged (after 0.5.0 release is done), and let's rebase this then on top of the master. Thanks for testing & patience :-)